server/src/main/java/facultad/trendz/service/PostService.java

```diff
@@ -124,4 +122,51 @@ public void deletePost(Long postId) {
                         comment.getUser().getId()))
                 .collect(Collectors.toList());
     }
+
+    public VoteResponseDTO upvotePost(Long userId, Long postId) {
+        final Optional<Post> post = postRepository.findById(postId);
+        if (!post.isPresent()) throw new PostNotFoundException();
+        final Optional<User> user = userRepository.findById(userId);
```

Comment:
throw UserNotFoundException if the user is not there, otherwise it will throw a null pointer